fortnite_api/cosmetics/common.py

```diff
@@ -0,0 +1,176 @@
+        self.display_value: str = data['displayValue']
+        self.backend_value: str = data['backendValue']
+
+
+class CosmeticRarity:
+    """Represents a cosmetic rarity.
+
+    Attributes
+    ----------
+    value: :class:`str`
+        The value of the cosmetic rarity.
+    display_value: :class:`str`
+        The display value of the cosmetic rarity. This is the value that is displayed to the user.
+    backend_value: :class:`str`
+        The internal marker of the cosmetic rarity.
+    """
+
+    __slots__: Tuple[str, ...] = ('value', 'display_value', 'backend_value')
+
+    def __init__(self, *, data: Dict[str, Any]) -> None:
+        self.value: str = data['value']
+        self.display_value: str = data['displayValue']
+        self.backend_value: str = data['backendValue']
+
+
+class CosmeticSeries(Generic[HTTPClientT]):
+    """Represents the series of a cosmetic.
+
+    Attributes
+    ----------
+    value: :class:`str`
+        The value of the cosmetic series.
+    backend_value: :class:`str`
+        The internal marker of the cosmetic series.
+    image: Optional[:class:`Asset`]
+        The image of the cosmetic series. Will be ``None`` if the cosmetic series has no image.
+    colors: List[:class:`str`]
+        A list of colors that are associated with the cosmetic series.
+    """
+
+    __slots__: Tuple[str, ...] = ('value', 'backend_value', 'image', 'colors')
+
+    def __init__(
+        self,
+        *,
+        data: Dict[str, Any],
+        http: HTTPClientT,
+    ) -> None:
+        self.value: str = data['value']
+        self.backend_value: str = data['backendValue']
+
+        image = data.get('image')
+        self.image: Optional[Asset[HTTPClientT]] = image and Asset(http=http, url=image)
+
+        self.colors: List[str] = get_with_fallback(data, 'colors', list)
+
+
+class CosmeticImages(Images[HTTPClientT]):
+    """Represents the images of a cosmetic.
+
+    This inherits from :class:`Images`.
+
+    Attributes
+    ----------
+    featured: Optional[:class:`Asset`]
+        The featured image of the cosmetic, if available.
+    lego: Optional[:class:`Asset`]
+        The LEGO image of the cosmetic, if available.
+    other: Dict[:class:`str`, :class:`Asset`]
+        A mapping other images to their respective asset.
+    """
+
+    __slots__: Tuple[str, ...] = ('featured', 'lego', 'other')
+
+    def __init__(
+        self,
+        *,
+        data: Dict[str, Any],
+        http: HTTPClientT,
+    ) -> None:
+        super().__init__(data=data, http=http)
+
+        featured = data.get('featured')
+        self.featured: Optional[Asset[HTTPClientT]] = featured and Asset(http=http, url=featured)
+
+        lego = data.get('lego')
+        self.lego: Optional[Asset[HTTPClientT]] = lego and Asset(http=http, url=lego)
+
+        # Mapping of str to Asset. In testing all I have found is "background" in this mapping.
+        other: Dict[str, str] = get_with_fallback(data, 'other', dict)
```

Comment:
We can be more specific here. Images include:
- background for loading screens
- coverart for lobby music
- decal for sprays

Reply:
By any chance can you provide the specific keys for this?

Reply:
Should this remain under the `other` attribute or become attributes of the CosmeticImages object itself?